saleor/product/utils/variant_prices.py

```diff
@@ -0,0 +1,49 @@
+from django.db.models.query_utils import Q
+
+from ...discount.utils import fetch_active_discounts
+from ..models import Product
+
+
+def _get_product_minimal_variant_price(product, discounts):
+    # Start with the product's price as the minimal one
+    minimal_variant_price = product.price
+    for variant in product.variants.all():
+        variant_price = variant.get_price(discounts=discounts)
+        minimal_variant_price = min(minimal_variant_price, variant_price)
+    return minimal_variant_price
+
+
+def update_product_minimal_variant_price(product, discounts=None):
+    if discounts is None:
+        discounts = fetch_active_discounts()
+    minimal_variant_price = _get_product_minimal_variant_price(product, discounts)
+    if product.minimal_variant_price != minimal_variant_price:
+        product.minimal_variant_price = minimal_variant_price
+        product.save(update_fields=["minimal_variant_price"])
+    return product
+
+
+def update_products_minimal_variant_prices(products, discounts=None):
+    if discounts is None:
+        discounts = fetch_active_discounts()
+    for product in products:
+        update_product_minimal_variant_price(product, discounts)
```

Comment:
Instead of making `.save()` after `.save()` we could do single `bulk_update` at the end of function. https://docs.djangoproject.com/en/2.2/ref/models/querysets/#bulk-update
It will be chaper.

Reply:
Good idea! Changed.